File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        HashSet<int> hashset = new HashSet<int>(nums);
        int max = 0;
        
        for(int i = 0; i<nums.Length; i++){
            int currMax = 1;
            int currNumber = nums[i];
            if(!hashset.Contains(currNumber - 1)) {
                currNumber = nums[i] + 1;
                while(hashset.Contains(currNumber)) {
                    currMax++;
                    currNumber += 1;
                }
                max = Math.Max(max, currMax);   
            }
        }
        
        return max;
    }
}

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        result = 0
        
        for value in nums:
            if value - 1 not in nums_set:
                curr_value = value
                curr_result = 0
                while curr_value in nums_set:
                    curr_value += 1
                    curr_result += 1
                result = max(result, curr_result)
            
        return result